### PR TITLE
Create HttpServer.h

### PR DESCRIPTION
**Pull request by ian287913**

### CODE REVIEW

File: src/HttpServer.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include <iostream>
```
Comment (john0312) [inline src/HttpServer.h:1]: Please add license header.

File: src/HttpServer.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#include "ThreadPool.h"
+
+using namespace std;
```
Comment (john0312) [inline src/HttpServer.h:4]: Please do not use "using namespace". Reference all classes by their full namespace path or add "using" in the class definition.
https://google.github.io/styleguide/cppguide.html#Namespaces

File: src/HttpServer.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+public class HttpServer
```
Comment (john0312) [inline src/HttpServer.h:6]: Please add class comment.
https://google.github.io/styleguide/cppguide.html#Class_Comments

File: src/HttpServer.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+{
+public:
+	HttpServer();
+	~HttpServer();
+    int serverStatus = 0;
```
Comment (john0312) [inline src/HttpServer.h:11]: Please make member variables private.
https://google.github.io/styleguide/cppguide.html#Access_Control

File: src/HttpServer.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#include <iostream>
+#include "ThreadPool.h"
+
+using namespace std;
+
+public class HttpServer
+{
```
Comment (john0312) [inline src/HttpServer.h:7]: Is this class copyable? If not, please delete its copy constructor and assignment operator.

File: src/HttpServer.h [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+public:
+	HttpServer();
+	~HttpServer();
+    int serverStatus = 0;
+    int clientCount = 0;
+    int maxClientsCount = 0;
+private:
+    void InitializeListenerThreadPool();
```
Comment (john0312) [inline src/HttpServer.h:15]: Please add function comment.
https://google.github.io/styleguide/cppguide.html#Function_Comments

File: src/HttpServer.h [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+// Example:
+//    std::unique_ptr<HttpServer> httpServer = make_unique<HttpServer>();
+//    // start the server
+//    httpServer->Start();
+public class HttpServer
+{
+public:
+    // Constructor
+	HttpServer();
+    // Deprecate copy constructors
```
Comment (john0312) [inline src/HttpServer.h:28]: Do you mean "Delete"?

File: src/HttpServer.h [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    HttpServer(const HttpServer&) = delete;
+    HttpServer& operator=(const HttpServer&) = delete;
+    // Destructor
+	~HttpServer();
+    // Start the http server
+    void Start();
+    // Close the http server
+    void Close();
```
Comment (john0312) [inline src/HttpServer.h:36]: Consider calling it "Shutdown"

**Comment by john0312** [comment]
Hello @destinycan (Zoan), please take a look at this commit/PR. Thank you.

**Comment by destinycan** [comment]
LGTM

**Pull request merged**